src/i18n/i18n-loader.js

```diff
@@ -65,7 +65,7 @@ const getLocale = (localeStr) => {
   // 2. User setting in cookie
   const cookieLanguagePreference = cookies.get(process.env.LANGUAGE_PREFERENCE_COOKIE_NAME);
   if (cookieLanguagePreference && messages[cookieLanguagePreference] !== undefined) {
-    return cookieLanguagePreference;
+    return cookieLanguagePreference.substr(0, 2);
```

Comment:
You could break this out into a `parseLanguageCode` function now that you are doing it twice. Or not.